From dc871bcf1d8dbb571d20b88356209100daa453c8 Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Tue, 6 Jan 2026 14:32:55 +0800
Subject: [PATCH] Linkage to _cygwin_dll_entry for mpc

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 23231da..a8bb176 100644
--- a/configure.ac
+++ b/configure.ac
@@ -213,6 +213,7 @@ AS_IF([test "x$enable_logging" = "xyes"],
      )
 
 # Configs for Windows DLLs
+MPC_LDFLAGS="$MPC_LDFLAGS -Wl,-e -Wl,_cygwin_dll_entry "
 AC_SUBST(MPC_LDFLAGS)
 case $host in
   *-*-cygwin* | *-*-mingw* | *-*-pw32* | *-*-os2*)
-- 
2.52.0.windows.1

