From 2eb94e9a33b482a2d37528f149128a9db1e11fed Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Mon, 29 Dec 2025 20:32:45 +0800
Subject: [PATCH 19/20] builtin_define MSYS and update t-cygwin-w64

---
 gcc/config/i386/cygwin.h     | 1 +
 gcc/config/i386/t-cygwin-w64 | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/gcc/config/i386/cygwin.h b/gcc/config/i386/cygwin.h
index c3267a12f8f..dff4435aa03 100644
--- a/gcc/config/i386/cygwin.h
+++ b/gcc/config/i386/cygwin.h
@@ -22,6 +22,7 @@ along with GCC; see the file COPYING3.  If not see
   do								\
     {								\
       builtin_define ("__CYGWIN__");				\
+      builtin_define ("__MSYS__");				\
       if (!TARGET_64BIT)					\
 	builtin_define ("__CYGWIN32__");			\
       builtin_define_std ("unix");				\
diff --git a/gcc/config/i386/t-cygwin-w64 b/gcc/config/i386/t-cygwin-w64
index e69de29bb2d..9116267ebc0 100644
--- a/gcc/config/i386/t-cygwin-w64
+++ b/gcc/config/i386/t-cygwin-w64
@@ -0,0 +1,3 @@
+MULTILIB_OPTIONS = m64/m32
+MULTILIB_DIRNAMES = 64
+MULTILIB_OSDIRNAMES = ../lib ../lib32
\ No newline at end of file
-- 
2.52.0.windows.1

