From a8877966795c40458faf2610fcebbe9672954891 Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Tue, 6 Jan 2026 05:47:28 +0800
Subject: [PATCH] Switch to use _cygwin_dll_entry for gmp

handle libtool
---
 configure.ac | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/configure.ac b/configure.ac
index 082077b..420d499 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2465,6 +2465,7 @@ if test $found_compiler = yes; then
   #
                             eval GMP_LDFLAGS=\"\$${ccbase}${abi1}_ldflags\"
   test -n "$GMP_LDFLAGS" || eval GMP_LDFLAGS=\"\$${ccbase}${abi1}_ldflags\"
+  GMP_LDFLAGS="$GMP_LDFLAGS -Wl,-e,_cygwin_dll_entry"
   AC_SUBST(GMP_LDFLAGS)
   AC_SUBST(LIBGMP_LDFLAGS)
   AC_SUBST(LIBGMPXX_LDFLAGS)
@@ -2535,6 +2536,12 @@ AC_PROG_CC
 AC_PROG_CC_C99
 AC_PROG_CPP
 
+# Enable libtool support
+AC_PROG_LIBTOOL
+# Configs for Windows DLLs.
+AC_LIBTOOL_WIN32_DLL
+LT_INIT
+
 #if test "$ac_cv_prog_cc_c99" = no; then
 #  AC_MSG_ERROR([Cannot find a C99 capable compiler])
 #fi
@@ -2720,11 +2727,6 @@ case $host in
     if test -z "$enable_shared"; then enable_shared=no; fi ;;
 esac
 
-
-# Configs for Windows DLLs.
-
-AC_LIBTOOL_WIN32_DLL
-
 AC_SUBST(LIBGMP_DLL,0)
 case $host in
   *-*-cygwin* | *-*-mingw* | *-*-msys | *-*-pw32* | *-*-os2*)
@@ -2813,7 +2815,6 @@ fi
 #
 #AC_PROVIDE([AC_PROG_F77])
 
-AC_PROG_LIBTOOL
 
 # Generate an error here if attempting to build both shared and static when
 # $libname.a is in $library_names_spec (as mentioned above), rather than
@@ -4061,8 +4062,6 @@ fi
 AC_PROG_YACC
 AM_PROG_LEX
 
-LT_INIT
-
 # Create config.m4.
 GMP_FINISH
 
-- 
2.52.0.windows.1

